lib/components/app/call-taker-panel.js

```diff
@@ -0,0 +1,677 @@
+      intermediatePlaces.push({})
+    }
+    this.props.setQueryParam({intermediatePlaces})
+  }
+
+  _removePlace = ({index}) => {
+    const intermediatePlaces = [...this.props.currentQuery.intermediatePlaces] || []
+    intermediatePlaces.splice(index, 1)
+    this.props.setQueryParam({intermediatePlaces})
+  }
+
+  _setMode = evt => {
+    const {value: mode} = evt.target
+    const transitIsSelected = mode.indexOf('TRANSIT') !== -1
+    if (transitIsSelected) {
+      // Collect transit modes and selected access mode.
+      const accessMode = mode === 'TRANSIT' ? 'WALK' : 'BICYCLE'
+      // If no transit is selected, selected all available. Otherwise, default
+      // to state.
+      const transitModes = this.state.transitModes.length > 0
+        ? this.state.transitModes
+        : this.props.modes.transitModes.map(m => m.mode)
+      const newModes = [accessMode, ...transitModes].join(',')
+      this.setState({transitModes})
+      this.props.setQueryParam({ mode: newModes })
+    } else {
+      this.props.setQueryParam({ mode })
+    }
+  }
+
+  _onHideAdvancedClick = () => {
+    const expandAdvanced = !this.state.expandAdvanced
+    // FIXME move logic to action
+    storeItem('expandAdvanced', expandAdvanced)
+    this.setState({expandAdvanced})
+  }
+
+  /**
+   * Key down listener that will submit a plan trip query when Enter is pressed.
+   * This listener can be passed to constituent form elements for easy
+   * keyboard-only operation of the trip planning form. Note: it generally should
+   * not be passed to buttons or other elements that natively rely on the Enter
+   * key.
+   */
+  _handleFormKeyDown = (evt) => {
+    switch (evt.keyCode) {
+      case 13: // Enter
+        evt.preventDefault()
+        // Submit routing query.
+        this._planTrip()
+        break
+      default:
+        // Do nothing.
+        break
+    }
+  }
+
+  render () {
+    const {
+      activeSearch,
+      datastoreUrl,
+      currentQuery,
+      itineraryFooter,
+      LegIcon,
+      mainPanelContent,
+      mobile,
+      modes,
+      ModeIcon,
+      routes,
+      setQueryParam,
+      showUserSettings
+    } = this.props
+    const showPlanTripButton = mainPanelContent === 'EDIT_DATETIME' ||
+      mainPanelContent === 'EDIT_SETTINGS'
+    // const mostRecentQuery = activeSearch ? activeSearch.query : null
+    // const planDisabled = isEqual(currentQuery, mostRecentQuery)
+    const {
+      departArrive,
+      date,
+      intermediatePlaces,
+      mode,
+      time
+    } = currentQuery
+    const actionText = mobile ? 'tap' : 'click'
+    const {expandAdvanced} = this.state
+    return (
+      <ViewerContainer>
+        <div className='main-panel' style={{
+          position: 'absolute',
+          top: 0,
+          left: 0,
+          right: 0,
+          bottom: showPlanTripButton ? 55 : 0,
+          paddingBottom: 15,
+          overflow: 'hidden',
+          padding: '10px'
+        }}>
```

Comment:
Should we start using styled-components instead of inlining styles like this?

Reply:
Dunno. The only reason I opted for inlining was because I'm not as familiar with styled-components. I'm happy for us to make a decision one way or another.